Orchestrate/Utility/RestClient.cs

```diff
@@ -0,0 +1,91 @@
+using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+
+namespace Orchestrate.Io.Utility
+{
+    public class RestClient
+    {
+        private readonly string apiKey;
+        private readonly JsonSerializer serializer;
+
+        public RestClient(string apiKey, JsonSerializer serializer)
```

Comment:
How intimately is the code tied to JSON.NET? Since you've gone to all this trouble, it would seem like it would be trivial (and useful) to abstract the serializer to make the entire thing pluggable, not just configurable.